octue/resources/datafile.py

```diff
@@ -93,6 +95,22 @@ def __init__(
         if not skip_checks:
             self.check(**kwargs)
 
+        self._gcp_metadata = None
+
+    @classmethod
+    def from_google_cloud_storage(cls, project_name, bucket_name, path_in_bucket, cluster=0, sequence=None):
+        """Create a Datafile from a file in Google Cloud storage."""
+        client = GoogleCloudStorageClient(project_name)
+
+        datafile = cls(
+            path=f"gs://{bucket_name}/{path_in_bucket}",
+            cluster=cluster,
+            sequence=sequence,
+        )
+
+        datafile._gcp_metadata = client.get_metadata(bucket_name, path_in_bucket)
```

Comment:
do cluster, sequence, tags attributes get populated at all?

Reply:
Cluster and sequence do just above but have to be specified (see [here](https://gitlab.com/windenergie-hsr/aerosense/digital-twin/data-gateway/-/merge_requests/5/diffs#83fd1db0589e993a988dcdbc0889ce54b16c623b_0_42))  although not tags currently.

Is it the custom metadata key-value pairs we could use as tags on google cloud?

Reply:
This now populates automatically if metadata is provided when files are uploaded to GCP.